### PR TITLE
10959-location-impression-events-bug

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
https://swrvedev.jira.com/browse/SWRVE-10959

@shanetmoore

### CODE REVIEW

**Comment by shanethmoore** [comment]
Will developers have to declare this wakeful service in their manifests?

**Comment by dominicmarmionswrve** [comment]
Similar to the ConversationActivity, it will be defined in our aar file, so they won't be required to define it in their manifest.

File: SwrveSDK/src/main/java/com/swrve/sdk/QueuedEventsManager.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    deleteEvents = false;
+                    SwrveLogger.e(LOG_TAG, "Error sending events to Swrve: " + response.responseBody);
+                }
+
+                // Resend if we got a server error (5XX)
+                listener.onResponse(deleteEvents);
+            }
+
+            @Override
+            public void onException(Exception exp) {
```
Comment (shanethmoore) [inline SwrveSDK/src/main/java/com/swrve/sdk/QueuedEventsManager.java]: I know this was copy/paste but while we're in the neighbourhood, can we throw a SwrveLogger.e in here?

**Comment by shanethmoore** [comment]
Awesome work man :+1: :rocket: :moon: Really happy we're finally pulling stuff out of SwrveIMP and SwrveBase

File: SwrveSDK/src/main/java/com/swrve/sdk/QueuedEventsManager.java (hunk 1 above)
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/QueuedEventsManager.java]: Yes, will do, good catch.

**Pull request merged**